client/coral-framework/store/actions/items.js

```diff
@@ -89,26 +78,48 @@ export const appendItemRelated = (item_id, property, value) => {
 * @dispatches
 *   A set of items to the item store
 */
-export function getItemsQuery (rootId) {
+export function getStream (assetId) {
   return (dispatch) => {
-    // return fetch('/v1/exec/view/' + rootId)
-    //   .then(
-    //     response => {
-    //       return response.ok ? response.json() : Promise.reject(response.status + ' ' + response.statusText)
-    //     }
-    //   )
-    //   .then((json) => {
-    //     let items = json.results[0].Docs
-    //     for (var i = 0; i < items.length; i++) {
-    //       dispatch(addItem(items[i]))
-    //     }
-    //     return (items)
-    //   })
-    console.log('Loading mock data', mocks);
-    let mockData = mocks.query
-    for (var i=0; i < mockData.length; i++ ) {
-      dispatch(addItem(mockData[i]))
-    }
+    return fetch('/api/v1/stream?asset_id='+assetId)
+      .then(
+        response => {
+          return response.ok ? response.json() : Promise.reject(response.status + ' ' + response.statusText)
+        }
+      )
+      .then((json) => {
+
+        /* Sort comments by date*/
+        let rootComments = []
+        let childComments = {}
+        const sorted = json.sort((a,b) => b.created_at - a.created_at)
+        sorted.reduce((prev, item) => {
```

Comment:
From @riley: since this block doesn't return things by keys, we should use a for each here.